gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/crud_service/access_point/AccessPointCrudServiceImpl.java

```diff
@@ -70,11 +94,23 @@ public void deleteAccessPoints(final AccessPoint.ReferenceType referenceType, fi
                 AccessPointReferenceType.valueOf(referenceType.name()),
                 referenceId
             );
+
             if (deleteAccessPoints != null) {
-                deleteAccessPoints
-                    .stream()
-                    .map(AccessPointAdapter.INSTANCE::toEntity)
-                    .forEach(accessPoint -> eventManager.publishEvent(AccessPointEvent.DELETED, accessPoint));
+                for (io.gravitee.repository.management.model.AccessPoint accessPoint : deleteAccessPoints) {
+                    eventManager.publishEvent(AccessPointEvent.DELETED, AccessPointAdapter.INSTANCE.toEntity(accessPoint));
+
+                    if (
+                        accessPoint.getTarget().equals(AccessPointTarget.GATEWAY) &&
+                        accessPoint.getReferenceType().equals(AccessPointReferenceType.ENVIRONMENT)
+                    ) {
+                        eventService.createAccessPointGatewayEvent(
+                            new ExecutionContext(null, accessPoint.getReferenceId()),
```

Comment:
Could you directly send the environments ID without creating a empty execution contxt?